src/org/traccar/BaseProtocolDecoder.java

```diff
@@ -31,6 +31,22 @@
 
     private final Protocol protocol;
 
+    public long addUnknownDevice(String uniqueId, SocketAddress remoteAddress) {
+        long deviceId = 0;
+        Device newDevice = new Device();
+        newDevice.setName(uniqueId);
+        newDevice.setUniqueId(uniqueId);
+
+        try {
+            Context.getDeviceManager().addDevice(newDevice);
+            Log.info("Nice meeting new device " + uniqueId + ".");
+        } catch (Exception e) {
```

Comment:
Moving log down was a bad idea. Now it means that even if device wasn't registered (SQL error), you would still log a message that it was registered.